discord/ext/commands/cooldowns.py

```diff
@@ -128,15 +124,19 @@ def reset(self):
         self._last = 0.0
 
     def copy(self):
-        return Cooldown(self.rate, self.per, self.type)
+        return Cooldown(self.rate, self.per)
 
     def __repr__(self):
         return f'<Cooldown rate: {self.rate} per: {self.per} window: {self._window} tokens: {self._tokens}>'
 
 class CooldownMapping:
-    def __init__(self, original):
+    def __init__(self, original, type):
+        if not isinstance(type, BucketType):
```

Comment:
```suggestion
        if not callable(type):
```

This is a regression.